src/oasis-nexus/api.ts

```diff
@@ -145,20 +145,26 @@ declare module './generated/api' {
 
 export const isAccountEmpty = (account: RuntimeAccount | Account) => {
   if (account.layer === Layer.consensus) {
-    // TODO: find a sane way to recognize an important consensus account.
-    // The heuristics below is clearly insufficient, because it would indicate even named accounts like "Governance Escrow" to be empty.
-    const { available, size, nonce, debonding_delegations_balance, delegations_balance, escrow, total } =
-      account as Account
+    const {
+      available,
+      nonce,
+      debonding_delegations_balance,
+      delegations_balance,
+      escrow,
+      total,
+      first_activity,
+      stats,
+    } = account as Account
     return (
       available === '0' &&
       debonding_delegations_balance === '0' &&
       delegations_balance === '0' &&
       escrow === '0' &&
       total === '0' &&
       nonce === 0 &&
-      size === 'XXS'
+      stats.num_txns === 0 &&
+      !first_activity
```

Comment:
I am not exactly clear of the definition of "activity", but doesn't this last check also cover everything else, making all other conditions redundant?

The API docs says

> The second-granular consensus time of the block in which this account was first active. Dates before Cobalt (2021-04-28) are approximate.

Are there cases when `first_activity` is missing, but there is still some activity we can detect from other fields? (I'll try to clarify this with backend.)

Reply:
It might cover everything but I'm not sure